nats.go

```diff
@@ -283,6 +283,10 @@ type CustomDialer interface {
 	Dial(network, address string) (net.Conn, error)
 }
 
+type InProcessConnProvider interface {
+	InProcessConn() (net.Conn, error)
```

Comment:
Following Jaime's recommendation, I think this could be simply `Conn()` here..

Reply:
Done.